File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left=0,right=heights.size()-1;
        int max_area=0;
        while(left<right){
            max_area=max( max_area,(right-left)*min(heights[left],heights[right]));
            if(heights[left]<heights[right]) left++;
            else right--;
        }
        return max_area;
        
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode],d=0) -> int:
        if not root:
            return d;
        a=self.maxDepth(root.left,d+1)
        b=self.maxDepth(root.right,d+1)
        return max(a,b)
